### PR TITLE
Avoid generating association if table object or database table exist

### LINKED ISSUE #839: RFC: Do not generate associations for `_id` fields if no Table object or database table present
### Description

When baking a model, if you database table has a field with `_id` suffix it generates a belongsTo association. It is usually useful but sometimes it is not an actual association.

i.e A `users` table with an `external_id` to store the user id in another app/service.

Currently Bake would generate a belongsTo association with `Externals` as alias, even if ExternalsTable does not exist and `externals` table does not exist onto database.

I would propose to only generate the association if any of the above conditions is met.

I have a partial solution already but I would like to know your thoughts.

### PR DESCRIPTION
**Pull request by ajibarra**
This PR fixes #839 and #834 . Please check and let me know if any change is needed.

Thank you

### CODE REVIEW

File: src/Command/ModelCommand.php [hunk 1]
```diff
@@ -346,11 +346,19 @@ public function findBelongsTo(Table $model, array $associations): array
                 ];
             } else {
                 $tmpModelName = $this->_modelNameFromKey($fieldName);
-                if (!in_array(Inflector::tableize($tmpModelName), $this->_tables, true)) {
+                $this->getTableLocator()->get($tmpModelName);
+                $genericInstances = $this->getTableLocator()->genericInstances();
+                $tables = $this->listAll();
+                /** Check if association model could not be instantiated as a subclass but a generic Table instance instead. */
```
Comment (othercorey) [inline src/Command/ModelCommand.php]:
```suggestion
                // Check if association model could not be instantiated as a subclass but a generic Table instance instead
```

File: src/Command/ModelCommand.php [hunk 2]
```diff
@@ -346,11 +346,19 @@ public function findBelongsTo(Table $model, array $associations): array
                 ];
             } else {
                 $tmpModelName = $this->_modelNameFromKey($fieldName);
-                if (!in_array(Inflector::tableize($tmpModelName), $this->_tables, true)) {
+                $this->getTableLocator()->get($tmpModelName);
+                $genericInstances = $this->getTableLocator()->genericInstances();
```
Comment (othercorey) [inline src/Command/ModelCommand.php]: Could also use get_class() to check the type returned by the above get() call.

File: src/Command/ModelCommand.php [hunk 3]
```diff
@@ -346,11 +346,18 @@ public function findBelongsTo(Table $model, array $associations): array
                 ];
             } else {
                 $tmpModelName = $this->_modelNameFromKey($fieldName);
-                if (!in_array(Inflector::tableize($tmpModelName), $this->_tables, true)) {
+                $associationTable = $this->getTableLocator()->get($tmpModelName);
+                $tables = $this->listAll();
```
Comment (markstory) [inline src/Command/ModelCommand.php:350]: You should be able to use `$this->_tables` here as it was used here before.

Comment (ADmad) [reply src/Command/ModelCommand.php:350]: listAll() returns the value of $_tables if it's already populated else fetches the list from the db. I think using the method is fine here. That way it doesn't need to rely on listAll() being called before this method is invoked.

Comment (ajibarra) [reply src/Command/ModelCommand.php:350]: That's correct. I initially used $this->_tables but found unit tests failed when findBelongsTo was called isolated. generateAssociations calls listAll as well as findHasMany and findBelongsToMany methods but findBelongsTo didn't.

**Review by ADmad — APPROVED** [review]

**Pull request merged**